core/src/main/java/com/yahoo/imapnio/async/request/AbstractQueryFoldersCommand.java

```diff
@@ -39,24 +48,25 @@ public void cleanup() {
     }
 
     @Override
-    public String getCommandLine() {
+    public ByteBuf getCommandLineBytes() throws ImapAsyncClientException {
         // Ex:LIST /usr/staff/jones ""
 
         // encode the arguments as per RFC2060
         final String ref64 = BASE64MailboxEncoder.encode(ref);
         final String pat64 = BASE64MailboxEncoder.encode(pattern);
```

Comment:
I remember according to imap rfc, this does not have to be imap utf-7 encoded. Please check.